### ISSUE TITLE
GHA SPIKE

### ISSUE THREAD

**Issue by non-fungible-nelson** [hyperledger/besu#6238]
https://wiki.hyperledger.org/pages/viewpage.action?pageId=80774216

Planned CI/CD Design
- Subdivide CI process into reusable phases, and defer long-running and expensive phases till as late and infrequently as possible, while still ensuring maximum potential quality checks.

On new PR open, draft or otherwise, the following phases can all be run, mostly in parallel and mostly on cheap, GitHub provided runners.

- compile all code
- [ ] then validate javadocs- this depends on bytecode output.
- [ ] check for repo compliance via repolinter
check for source code formatting via spotless
check gradle tooling validation.
Devs are expected to have run as many tests as feasible before review. Once the checks above have passed for sanity:

run unitTests
any test failures will be annotated on the PR.
tests will be sharded based on prior runtimes.
Once the PR is approved, the following can be run in parallel.

run acceptanceTestsPermissioning
run acceptanceTestsCliqueBFT
run integrationTests
run acceptanceTests
run referenceTests
Test results can then be posted back to the PR, and any failures would prevent merging. If all pass, then the following artifacts are created, unique to the commit for this workflow.

tarball created
zipfile created
docker images created
Should a release be decided on for this commit, the creation of a draft release would trigger:

publishing of prior created artifacts with the release name and an RC1 suffix.
creation of a release branch, should additions be necessary for an RC2.
When a release comes out of draft, and is approved:

all artifacts are published as a non RC version number.
docker latest tag is moved.
